internal/provider/aws/provider.go

```diff
@@ -766,28 +781,26 @@ func (p *awsProvider) ReconcileFirewallRule(
 			return status, fmt.Errorf("failed to modify network interface attribute: %w", err)
 		}
 		log.Info("Security group disassociated from network interface", "securityGroupID", securityGroupID, "networkInterfaceID", ni.NetworkInterfaceId)
-
-		p.networkInterfacesCache.Delete(securityGroupID)
-
-		p.securityGroupsCache.Flush()
-		p.securityGroupsMutex.Unlock()
 	}
 
 	if !isAssociated {
 		// Associate the security group with the network interface
-		// Disassociate the security group from other network interfaces
 		groups := []string{}
 		for _, group := range networkInterface.Groups {
 			if aws.ToString(group.GroupId) != securityGroupID {
 				groups = append(groups, aws.ToString(group.GroupId))
 			}
 		}
 		groups = append(groups, securityGroupID)
-		p.securityGroupsMutex.Lock()
-		_, err = p.ec2.ModifyNetworkInterfaceAttribute(ctx, &ec2.ModifyNetworkInterfaceAttributeInput{
-			NetworkInterfaceId: networkInterface.NetworkInterfaceId,
-			Groups:             groups,
-		})
+
+		// Here we associate the security group to the network interface of the instance
+		// since networkInterfacesCache store the security group ID as key and the network interfaces the securitygroup
+		// is associated with as value, we need to remove the security group ID from the cache
+		// Also the groups of the network interface of the instance change here, so we need to remove the instanceID
+		// from the instancesCache
+		// NB: The security group does not hold any interfaceID in his structure
```

Comment:
I think this part should be commented above modifyNetworkInterfaceAttribute function, since it is the common of the wrapper:
> since networkInterfacesCache store the security group ID as key and the network interfaces the securitygroup is associated with as value, we need to remove the security group ID from the cache